### PR TITLE
Perf/#222 improve performance of family tree

### PR DESCRIPTION
**Pull request by Kwandes**
### This pull request closes:

✔️ Closes#222

### Which service(s) does this issue affect:

- [ ] 🛰️ API
- [x] 🛒 webstore
- [ ] 🎓 admin-page

### Types of changes

- [ ] 🐛 Bug fix (a non-breaking change which fixes an issue)
- [ ] 🧱 New feature (a non-breaking change which adds functionality)
- [x] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🤖 CI/CD (a change to the CI/CD pipeline)
- [x] ⚗️ Refactoring (an update to some already existing code)

### Documentation Updates

- [x] 📂 No need for updates
- [ ] 📁 Requires an update ( _not done within this issue_ )
- [ ] 🗃️ Has been updated ( _done within this issue_ )

### Testing checklist

- [x] 💪 Manual tests
- [x] 🔧 Automatic tests

#### Browser compatibility - _**[**Frontend only**]**_

- [x] Tested on Safari
- [x] Tested on Chrome
- [x] Tested on Mozilla Firefox

### What changes have been done within this issue?

- Changed family tree design canvas resolution 4K x 4k -> 2K x 2K. INcludes the miniature
- Change how the background is rendered (now only boxes and banners get rendered on every frame, huge improvement!)
- Changed fps 30 -> 60
- Adjust text scaling to match the new resolution
- Remove some console logs and refactor code

#### BREAKING CHANGE
Old designs, including orders people made, will now appear broken when viewed due to the positions being saved for 4k. Need confirmation that that is okay (loss of data). Solution options: manually convert old stuff, or add a system that will save position based on % from the edge (right now it uses pixels)

### How should this be manually tested?

Navigate to the products page and move the boxes around. Shit should work smoothly

### CODE REVIEW

File: apps/webstore/src/app/shared/components/products/family-tree/family-tree-design/family-tree-design.component.ts [hunk 1]
```diff
@@ -88,8 +88,8 @@ export class FamilyTreeDesignComponent
   public context: CanvasRenderingContext2D;
 
   canvasResolution = {
-    height: 4000,
-    width: 4000,
+    height: 1000,
+    width: 1000,
   };
```
Comment (Teodor25) [inline apps/webstore/src/app/shared/components/products/family-tree/family-tree-design/family-tree-design.component.ts:90]: This resolution is too low. It decreases the screenshot quality so the text becomes less visible on the print. If we have the design res be 4x4 k while the design is imutable, and 1x1 k whilte mutable, that would work. But i need to be able to get a higher resolution screenshot of the design.

**Review by Teodor25 — CHANGES_REQUESTED** [review]

File: apps/webstore/src/app/shared/components/products/family-tree/family-tree-design/family-tree-design.component.ts (hunk 1 above)
Comment (DryDish) [reply apps/webstore/src/app/shared/components/products/family-tree/family-tree-design/family-tree-design.component.ts:90]: I say lets test the canvas export tools before we gimp the users. svg export from canvas.

Comment (Kwandes) [reply apps/webstore/src/app/shared/components/products/family-tree/family-tree-design/family-tree-design.component.ts:90]: Automated export of the image into an SVG is a different issue. This issue and pull request is specifically about improving the performance of the current system

File: apps/webstore/src/app/shared/components/products/family-tree/family-tree-design/family-tree-design.component.ts [hunk 2]
```diff
@@ -107,7 +107,7 @@ export class FamilyTreeDesignComponent
 
   // render loop
   timeInterval;
-  framesPerSecond = 30; // FPS of the render loop
+  framesPerSecond = 45; // FPS of the render loop
```
Comment (DryDish) [inline apps/webstore/src/app/shared/components/products/family-tree/family-tree-design/family-tree-design.component.ts]: why is this not 60?

**Review by Teodor25 — COMMENTED** [review]
Try to reset the canvas size to be 4k, if it slows the performance too much, this is good PR is approved ;)

**Pull request merged**